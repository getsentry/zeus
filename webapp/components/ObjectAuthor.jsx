import React, {Component} from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import idx from 'idx';
import Gravatar from 'react-gravatar';
import MdPerson from 'react-icons/lib/md/person';

const Avatar = styled.span`
  img {
    border-radius: 2px;
  }

  img,
  svg {
    margin-right: 5px;
    display: inline-block;
    vertical-align: text-top;
  }
`;

export default class ObjectAuthor extends Component {
  static propTypes = {
    data: PropTypes.shape({
      author: PropTypes.shape({
        name: PropTypes.string,
        email: PropTypes.email
      }),
      source: PropTypes.shape({
        author: PropTypes.shape({
          name: PropTypes.string,
          email: PropTypes.email
        })
      })
    }).isRequired
  };

  render() {
    let {data} = this.props;
    let author = idx(data, _ => _.source.author) || data.author;
    if (!author || !(author.name || author.email)) return null;
    return (
      <span>
        <Avatar>
          {author.email ? (
            <Gravatar email={author.email} size={16} />
          ) : (
            <MdPerson size="16" />
          )}
        </Avatar>
        {author.name || author.email.split('@', 1)[0]}
      </span>
    );
  }
}
