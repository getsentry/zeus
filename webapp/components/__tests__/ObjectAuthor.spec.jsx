import React from 'react';
import {shallow} from 'enzyme';

import ObjectAuthor from '../ObjectAuthor';

describe('ObjectAuthor', () => {
  it('renders source author', () => {
    const tree = shallow(
      <ObjectAuthor
        data={{
          source: {
            author: {
              email: 'foo@example.com',
              name: 'Foo Bar'
            }
          }
        }}
      />
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders source anonymous', () => {
    const tree = shallow(
      <ObjectAuthor
        data={{
          source: {
            author: null
          }
        }}
      />
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders author', () => {
    const tree = shallow(
      <ObjectAuthor
        data={{
          author: {
            email: 'foo@example.com',
            name: 'Foo Bar'
          }
        }}
      />
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders anonymous', () => {
    const tree = shallow(
      <ObjectAuthor
        data={{
          author: null
        }}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
