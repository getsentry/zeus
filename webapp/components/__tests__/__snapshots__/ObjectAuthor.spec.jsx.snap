// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObjectAuthor renders anonymous 1`] = `""`;

exports[`ObjectAuthor renders author 1`] = `
<span>
  <styled.span>
    <Gravatar
      default="retro"
      email="foo@example.com"
      protocol="//"
      rating="g"
      size={16}
    />
  </styled.span>
  Foo Bar
</span>
`;

exports[`ObjectAuthor renders source anonymous 1`] = `""`;

exports[`ObjectAuthor renders source author 1`] = `
<span>
  <styled.span>
    <Gravatar
      default="retro"
      email="foo@example.com"
      protocol="//"
      rating="g"
      size={16}
    />
  </styled.span>
  Foo Bar
</span>
`;
